### PR TITLE
Fix check token NPE

### PR DESCRIPTION
**Pull request by pigxcloud**
Solve the problem of RemoteTokenServices.check_token NullPointerException.

If the auth-server cannot return correctly , postForMap will be empty

image

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@pigxcloud Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security-oauth&pullRequestId=1841)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-security-oauth&pullRequestId=1841) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@pigxcloud Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security-oauth&pullRequestId=1841)!

File: spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java [hunk 1]
```diff
@@ -109,6 +109,13 @@ public OAuth2Authentication loadAuthentication(String accessToken) throws Authen
 		headers.set("Authorization", getAuthorizationHeader(clientId, clientSecret));
 		Map<String, Object> map = postForMap(checkTokenEndpointUrl, formData, headers);
 
+		if (map == null) {
+			if (logger.isDebugEnabled()) {
+				logger.debug("check_token returned empty: " + accessToken);
```
Comment (rwinch) [inline spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java:114]: We should not log credentials (accessToken) as this leaks data.

Comment (pigxcloud) [reply spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java]: > We should not log credentials (accessToken) as this leaks data.

ok.  I have removed

Comment (rwinch) [reply spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java]: Thanks! I assigned to @jgrandja who will help guide you with getting this merged

File: spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java [hunk 2]
```diff
@@ -109,6 +109,13 @@ public OAuth2Authentication loadAuthentication(String accessToken) throws Authen
 		headers.set("Authorization", getAuthorizationHeader(clientId, clientSecret));
 		Map<String, Object> map = postForMap(checkTokenEndpointUrl, formData, headers);
 
+		if (map == null) {
```
Comment (jgrandja) [inline spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java:112]: Can you change this to `org.springframework.util.CollectionUtils.isEmpty()`

Comment (pigxcloud) [reply spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java]: > 你能改变成 `org.springframework.util.CollectionUtils.isEmpty()`

Ok, it has been modified and added test

**Comment by jgrandja** [comment]
Thanks for the PR @pigxcloud. This is now in master and 2.3.x.

**Pull request closed without merging**